commands/api/api.go

```diff
@@ -0,0 +1,543 @@
+
+			$ glab api issues --paginate
+
+			$ glab api graphql -f query='
+			  query {
+			    project(fullPath: "gitlab-org/gitlab-docs") {
+			      name
+			      forksCount
+			      statistics {
+			        wikiSize
+			      }
+			      issuesEnabled
+			      boards {
+			        nodes {
+			          id
+			          name
+			        }
+			      }
+			    }
+			  }
+			'
+
+			$ glab api graphql --paginate -f query='
+			  query($endCursor: String) {
+			    project(fullPath: "gitlab-org/graphql-sandbox") {
+			      name
+			      issues(first: 2, after: $endCursor) {
+			        edges {
+			          node {
+			            title
+			          }
+			        }
+			        pageInfo {
+			          endCursor
+			          hasNextPage
+			        }
+			      }
+			    }
+			  }'
+		`),
+		Annotations: map[string]string{
+			"help:environment": heredoc.Doc(`
+				GITLAB_TOKEN, OAUTH_TOKEN (in order of precedence): an authentication token for API requests.
+				GITLAB_HOST, GITLAB_URI, GITLAB_URL: specify a GitLab host to make request to.
+			`),
+		},
+		Args: cobra.ExactArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts.RequestPath = args[0]
+			opts.RequestMethodPassed = c.Flags().Changed("method")
+			opts.Config, _ = f.Config()
+
+			if c.Flags().Changed("hostname") {
+				if err := glinstance.HostnameValidator(opts.Hostname); err != nil {
+					return &cmdutils.FlagError{Err: fmt.Errorf("error parsing --hostname: %w", err)}
+				}
+			}
+
+			if opts.Paginate && !strings.EqualFold(opts.RequestMethod, "GET") && opts.RequestPath != "graphql" {
+				return &cmdutils.FlagError{Err: errors.New(`the '--paginate' option is not supported for non-GET requests`)}
+			}
+			if opts.Paginate && opts.RequestInputFile != "" {
+				return &cmdutils.FlagError{Err: errors.New(`the '--paginate' option is not supported with '--input'`)}
+			}
+
+			if runF != nil {
+				return runF(&opts)
+			}
+			return apiRun(&opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.Hostname, "hostname", "", "The GitLab hostname for the request (default is \"gitlab.com\" or authenticated host in current git directory)")
+	cmd.Flags().StringVarP(&opts.RequestMethod, "method", "X", "GET", "The HTTP method for the request")
+	cmd.Flags().StringArrayVarP(&opts.MagicFields, "field", "F", nil, "Add a parameter of inferred type")
+	cmd.Flags().StringArrayVarP(&opts.RawFields, "raw-field", "f", nil, "Add a string parameter")
+	cmd.Flags().StringArrayVarP(&opts.RequestHeaders, "header", "H", nil, "Add an additional HTTP request header")
+	cmd.Flags().BoolVarP(&opts.ShowResponseHeaders, "include", "i", false, "Include HTTP response headers in the output")
+	cmd.Flags().BoolVar(&opts.Paginate, "paginate", false, "Make additional HTTP requests to fetch all pages of results")
+	cmd.Flags().StringVar(&opts.RequestInputFile, "input", "", "The file to use as body for the HTTP request")
+	cmd.Flags().BoolVar(&opts.Silent, "silent", false, "Do not print the response body")
```

Comment:
Could we maybe add a flag for not pretty printing? This would be usefull for pipeing the output to something else like 
```sh
glab api /projects | jq `.foo`
```

Reply:
`glab` should automatically detect when pipeing output and on non-ttys. Tested with these and properly working
```sh
$ glab api /projects | tee file.txt

$ glab api /projects > file.txt

$ glab api /projects | jq -c > '.foo'

$ glab api /projects | jq > '.foo'
```

Reply:
Nice! Didn't know this.